codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoderTest.java

```diff
@@ -272,26 +272,30 @@ public void dataReadForUnknownStreamShouldApplyFlowControlAndFail() throws Excep
         }
     }
 
+    @Test(expected = Http2Exception.class)
+    public void dataReadForUnknownStreamThatNeverExistedShouldThrow() throws Exception {
+        when(connection.streamMayHaveExisted(STREAM_ID)).thenReturn(false);
+        when(connection.stream(STREAM_ID)).thenReturn(null);
+        final ByteBuf data = dummyData();
+        decode().onDataRead(ctx, STREAM_ID, data, 0, true);
```

Comment:
@shorea do we need to call `data.release()` in a finally block here ?

Reply:
@shorea ping me once you addressed it.

Reply:
@normanmaurer  just pushed a fix for this.